cdc/redo/writer/writer.go

```diff
@@ -99,72 +105,79 @@ type LogWriter struct {
 }
 
 // NewLogWriter creates a LogWriter instance. It is guaranteed only one LogWriter per changefeed
-// TODO: delete log files when changefeed removed, metric
 func NewLogWriter(ctx context.Context, cfg *LogWriterConfig) (*LogWriter, error) {
-	var errInit error
-	// currently, caller do not have the ability of self-healing, like try to fix if on some error,
-	// so initOnce just literary init once, do not support re-init if fail on some condition
-	initOnce.Do(func() {
-		if cfg == nil {
-			errInit = cerror.WrapError(cerror.ErrRedoConfigInvalid, errors.New("LogWriterConfig can not be nil"))
-			return
-		}
+	if cfg == nil {
+		return nil, cerror.WrapError(cerror.ErrRedoConfigInvalid, errors.New("LogWriterConfig can not be nil"))
+	}
 
-		rowCfg := &FileWriterConfig{
-			Dir:               cfg.Dir,
-			ChangeFeedID:      cfg.ChangeFeedID,
-			CaptureID:         cfg.CaptureID,
-			FileType:          common.DefaultRowLogFileType,
-			CreateTime:        cfg.CreateTime,
-			MaxLogSize:        cfg.MaxLogSize,
-			FlushIntervalInMs: cfg.FlushIntervalInMs,
-			S3Storage:         cfg.S3Storage,
-			S3URI:             cfg.S3URI,
-		}
-		ddlCfg := &FileWriterConfig{
-			Dir:               cfg.Dir,
-			ChangeFeedID:      cfg.ChangeFeedID,
-			CaptureID:         cfg.CaptureID,
-			FileType:          common.DefaultDDLLogFileType,
-			CreateTime:        cfg.CreateTime,
-			MaxLogSize:        cfg.MaxLogSize,
-			FlushIntervalInMs: cfg.FlushIntervalInMs,
-			S3Storage:         cfg.S3Storage,
-			S3URI:             cfg.S3URI,
-		}
-		logWriter = &LogWriter{
-			cfg: cfg,
-		}
-		logWriter.rowWriter, errInit = NewWriter(ctx, rowCfg)
-		if errInit != nil {
-			return
-		}
-		logWriter.ddlWriter, errInit = NewWriter(ctx, ddlCfg)
-		if errInit != nil {
-			return
+	initLock.Lock()
+	defer initLock.Unlock()
+
+	if v, ok := logWriters[cfg.ChangeFeedID]; ok {
+		// if cfg changed or already closed need create a new LogWriter
+		if cfg.String() == v.cfg.String() && !v.isStopped() {
+			return v, nil
 		}
+	}
```

Comment:
In which scenario will cfg changed and create a new redo log writer? I can think of one
`pause changefeed -> changefeed update config -> resume changefeed`, is there any other scenario?

Reply:
yes, like change file size, flush interval, or delete changefeed